verifier/src/report_body.rs

```diff
@@ -110,50 +122,125 @@ impl<T> GreaterThanEqualVerifier<T> {
     }
 }
 
-/// Verifier for ensuring [`Attributes`] values are equivalent.
-pub type AttributesVerifier = EqualityVerifier<Attributes>;
-impl IntoVerificationError for Attributes {
-    fn into_verification_error(expected: Self, actual: Self) -> VerificationError {
-        VerificationError::AttributeMismatch { expected, actual }
-    }
+/// Macro to generate types and traits used for types that only need `expected`
+/// and `actual` in the output data
+///
+/// # Arguments
+/// * `type_to_verify` - The type to be verified
+/// * `verifier` - The name to use for the verifier type
+/// * `verification_output_type` - The type that will be returned in the
+///   [`VerificationResult`]
+macro_rules! expected_actual_output_impls {
+    ($($type_to_verify:ty, $verifier:ident, $verification_output_type:ident;)*) => {$(
+        #[derive(Debug, Clone, Eq, PartialEq)]
```

Comment:
I may have cranked the macroize things up to 11 here, so may want to be more explicit or generic in a few of these items that got macroized